tests/integration-tests/tests/test_raid.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import logging
+
+import pytest
+
+from assertpy import assert_that
+from remote_command_executor import RemoteCommandExecutor
+from tests.common.schedulers_common import get_scheduler_commands
+from tests.storage.storage_common import verify_directory_correctly_shared
+
+
+@pytest.mark.regions(["us-east-1", "eu-west-1", "cn-north-1", "us-gov-west-1"])
+@pytest.mark.instances(["c5.xlarge"])
+@pytest.mark.schedulers(["sge", "awsbatch"])
+@pytest.mark.usefixtures("region", "os", "instance")
+def test_raid_performance_mode(scheduler, pcluster_config_reader, clusters_factory):
+    cluster_config = pcluster_config_reader()
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+
+    scheduler_commands = get_scheduler_commands(scheduler, remote_command_executor)
+    mount_dir = "/raid_dir"
+    _test_raid_correctly_configured(remote_command_executor, raid_type="0", volume_size=75, raid_devices=5)
+    _test_raid_correctly_mounted(remote_command_executor, mount_dir, volume_size=74)
+    _test_raid_correctly_shared(remote_command_executor, mount_dir, scheduler_commands)
+
+
+@pytest.mark.regions(["us-east-1", "eu-west-1", "cn-north-1", "us-gov-west-1"])
+@pytest.mark.instances(["c5.xlarge"])
+@pytest.mark.schedulers(["sge", "awsbatch"])
+@pytest.mark.usefixtures("region", "os", "instance")
+def test_raid_fault_tolerance_mode(scheduler, pcluster_config_reader, clusters_factory):
+    cluster_config = pcluster_config_reader()
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+
+    scheduler_commands = get_scheduler_commands(scheduler, remote_command_executor)
+    mount_dir = "/raid_dir"
+    _test_raid_correctly_configured(remote_command_executor, raid_type="1", volume_size=20, raid_devices=2)
+    _test_raid_correctly_mounted(remote_command_executor, mount_dir, volume_size=20)
+    _test_raid_correctly_shared(remote_command_executor, mount_dir, scheduler_commands)
+
+
+def _test_raid_correctly_mounted(remote_command_executor, mount_dir, volume_size):
+    logging.info("Testing raid {0} is correctly mounted".format(mount_dir))
+    result = remote_command_executor.run_remote_command(
+        "df -h -t ext4 | tail -n +2 | awk '{{print $2, $6}}' | grep '{0}'".format(mount_dir)
```

Comment:
Why are you using double braces for the awk command?

Reply:
that's to escape curly braces in format command: https://docs.python.org/3/library/string.html#format-string-syntax